plugins/serializers/carbon2/carbon2.go

```diff
@@ -3,32 +3,32 @@ package carbon2
 import (
 	"bytes"
 	"fmt"
-	"github.com/influxdata/telegraf"
 	"strconv"
 	"strings"
+
+	"github.com/influxdata/telegraf"
 )
 
-type serializer struct {
+type Serializer struct {
```

Comment:
Why do you make it public? Isn't enough that `NewSerializer` is public?

Reply:
That is to allow the type switch in here https://github.com/SumoLogic/telegraf/pull/1/files#r471975704 so that we warn users that they use an unsupported metrics format.